java/com/google/flatbuffers/Table.java

```diff
@@ -239,6 +243,28 @@ protected static int compareStrings(int offset_1, int offset_2, ByteBuffer bb) {
     if (len_1 > len_2) return 1;
     return 0;
   }
+  
+  /**
+   * Compare string from the buffer with the 'String' object.
+   *
+   * @param offset_1 An 'int' index of the first string into the bb.
+   * @param key Second string.
+   * @param bb A {@code ByteBuffer} to get the first string.
+   */
+  protected static int compareStrings(int offset_1, String key, ByteBuffer bb) {
+    offset_1 += bb.getInt(offset_1);
+    int len_1 = bb.getInt(offset_1);
+    int len_2 = key.length();
+    int startPos_1 = offset_1 + Constants.SIZEOF_INT;
+    int len = Math.min(len_1, len_2);
+    for (int i = 0; i < len; i++) {
+      if (bb.array()[i + startPos_1] != key.charAt(i))
```

Comment:
I guess for now you can leave the loop, as it seems Arrays.equals doesn't take offsets. Converting String to UTF-8 should happen before the loop in LookupByKey.